runelite-client/src/main/java/net/runelite/client/plugins/inventoryviewer/InventoryViewerOverlay.java

```diff
@@ -45,31 +49,59 @@
 import net.runelite.client.game.SpriteManager;
 import net.runelite.client.ui.overlay.Overlay;
 import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.ComponentConstants;
 import net.runelite.client.ui.overlay.components.ComponentOrientation;
 import net.runelite.client.ui.overlay.components.ImageComponent;
 import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
 
 class InventoryViewerOverlay extends Overlay
 {
 	private static final int INVENTORY_SIZE = 28;
+	private static final int INVENTORY_WIDTH = 4;
+	private static final Point SPRITE_PADDING = new Point(6, 4);
+	private static final int WRAPPER_MINIMUM_WIDTH = Constants.ITEM_SPRITE_WIDTH + SPRITE_PADDING.x * 2;
 	private static final ImageComponent PLACEHOLDER_IMAGE = new ImageComponent(
 		new BufferedImage(Constants.ITEM_SPRITE_WIDTH, Constants.ITEM_SPRITE_HEIGHT, BufferedImage.TYPE_4BYTE_ABGR));
 
 	private final Client client;
 	private final ItemManager itemManager;
 	private final InventoryViewerConfig config;
 
-	private final PanelComponent panelComponent = new PanelComponent();
+	private final PanelComponent wrapperComponent = new PanelComponent();
+	private final PanelComponent inventoryComponent = new PanelComponent();
+	private final TitleComponent freeSlotsComponent = TitleComponent.builder().build();
+
+	@Setter(AccessLevel.PACKAGE)
+	private Map<Integer, Integer> groupedItems = null;
+
+	@Setter(AccessLevel.PACKAGE)
+	private int remainingSpaces = -1;
 
 	private ImageComponent inventoryIconSprite;
 
 	@Inject
 	private InventoryViewerOverlay(Client client, ItemManager itemManager, SpriteManager spriteManager, InventoryViewerConfig config)
 	{
 		setPosition(OverlayPosition.BOTTOM_RIGHT);
-		panelComponent.setWrapping(4);
-		panelComponent.setGap(new Point(6, 4));
-		panelComponent.setOrientation(ComponentOrientation.HORIZONTAL);
+		inventoryComponent.setWrapping(INVENTORY_WIDTH);
+		inventoryComponent.setGap(SPRITE_PADDING);
+		inventoryComponent.setOrientation(ComponentOrientation.HORIZONTAL);
+		inventoryComponent.setBackgroundColor(null);
+		inventoryComponent.setBorder(new Rectangle(
+			ComponentConstants.STANDARD_BORDER,
+			0,
+			0,
+			ComponentConstants.STANDARD_BORDER));
+
+		wrapperComponent.setOrientation(ComponentOrientation.VERTICAL);
+		wrapperComponent.setWrapping(2);
+		wrapperComponent.setBorder(new Rectangle(
+			ComponentConstants.STANDARD_BORDER,
+			ComponentConstants.STANDARD_BORDER,
+			ComponentConstants.STANDARD_BORDER,
+			ComponentConstants.STANDARD_BORDER));
```

Comment:
This is the default border that PanelComponent sets, so it's not necessary to define it as such here.